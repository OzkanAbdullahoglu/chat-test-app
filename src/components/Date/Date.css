.list-item-date-root {
    width: 100%;
    display: flex;
    position: relative;
    box-sizing: border-box;
    text-align: left;
    align-items: center;
    padding-top: 8px;
    padding-bottom: 8px;
    justify-content: center;
    padding: 4px 16px;
    margin-top: 4px;
    margin-bottom: 4px; 
}

.list-item-date-text-primary {
     background-color: #d9f7ff;
     border-radius: 8px;
     padding: 4px 8px;
     display: flex;
}

.timestamp {
    background-color: #c8c5c5;
    flex: inherit;
    border-radius: 8px;
}
