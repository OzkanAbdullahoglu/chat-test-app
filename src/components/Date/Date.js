/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';
import './Date.css';
const Date = ({
  pack,
}) => (
  <div className="list-item-date-root">
    <div className="list-item-date-text-primary timestamp">
      <span>{pack.date}</span>
    </div>
  </div>
);

Date.propTypes = {
  pack: PropTypes.object,
};

export default Date;
