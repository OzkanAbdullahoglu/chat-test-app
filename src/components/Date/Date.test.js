/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import Date from './Date';

describe('Date', () => {
  const props = {
    pack: {
      id: 1,
      direction: 'in',
      status: 'read',
      timestamp: '1577834102',
      text: 'Alice was beginning',
      date: 'January 01,2020',
      time: '02:15',
      height: 45,
    },
  };

  it('renders without crashing', () => {
    const date = shallow(<Date {...props} />);
    expect(toJson(date)).toMatchSnapshot();
  });
  it('renders proper date from props', () => {
    const date = shallow(<Date {...props} />);
    expect(date.find('span').text()).toBe(props.pack.date);
  });
});
