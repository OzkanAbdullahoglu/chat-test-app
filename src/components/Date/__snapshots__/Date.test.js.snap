// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date renders without crashing 1`] = `
<div
  className="list-item-date-root"
>
  <div
    aria-label="Date"
    className="list-item-date-text-primary timestamp"
  >
    <span>
      January 01,2020
    </span>
  </div>
</div>
`;
